blazar/db/sqlalchemy/utils.py

```diff
@@ -293,6 +294,49 @@ def get_reservation_allocations_by_device_ids(device_ids, start_date, end_date,
     return reservations
 
 
+def get_recent_non_pending_reservation_by_host_id(host_id):
+    """returns the recent host reservation which not in
+    pending status and start_date of the reservation is less than current date
+
+    Args:
+        host_id (): Host id - primary key of ComputeHost table
+    """
+    session = get_session()
+    curr_date = datetime.now()
```

Comment:
can you add some time (maybe ~3-5 minutes) to this to ensure a lease that is about to start won't start while the cleanup is running?